ui/app/AppLayouts/Chat/controls/community/PermissionsDropdown.qml

```diff
@@ -0,0 +1,202 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14
+import QtQuick.Layouts 1.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+
+import shared.panels 1.0
+
+
+StatusDropdown {
+    id: root
+
+    property int mode: PermissionsDropdown.Mode.Add
+    property int initialPermissionType: PermissionTypes.Type.None
+
+    enum Mode {
+        Add, Update
+    }
+
+    signal done(int permissionType, string title, string asset)
+
+    width: d.width
+    padding: d.padding
+
+    // force keeping within the bounds of the enclosing window
+    margins: 0
+
+    onOpened: {
```

Comment:
`onAboutToShow`, you want this to happen _prior_ opening the dialog

Reply:
there is no observable difference, but yes, `onAboutToShow` seems to be better choice